### ISSUE TITLE
Invalid DST.ADDR for IPv6 destination

### ISSUE THREAD

**Issue by Kaciras** [JoshGlazebrook/socks#99]
In version 2.8.1, when connecting to an IPv6 destination, the request may have invalid length of the `DST.ADDR` part.

Reproduction:

```javascript
const net = require("net");
const { SocksClient } = require("socks");

const server = net.createServer(s => {
    let handshake;
    s.on("data", data => {
        if (!handshake) {
            handshake = data;
            s.write(new Uint8Array([5, 0]));
        } else {
            console.log(data);
        }
    });
});

server.listen();
server.on("listening", async () => {
    const { address, port } = server.address();
    await SocksClient.createConnection({
        command: "connect",
        proxy: {
            type: 5,
            host: address,
            port,
        },
        destination: {
            host: "::1",
            port: 54321,
        },
    });
});
```

The request message is `05 01 00 04 01 d4 31`, `::1` is serialized as one byte, but 16 bytes are expected.

**Comment by JoshGlazebrook** [comment]
This is fixed in 2.8.2

**Comment by Kaciras** [comment]
This change was not included in 2.8.2, which seems to have been forgotten to compile before publish.

**Comment by JoshGlazebrook** [comment]
My mistake. I published 2.8.3
